test/controllers/foreign/allowances/ForeignAllowancesCheckYourAnswersControllerSpec.scala

```diff
@@ -0,0 +1,152 @@
+
+import audit.AuditService
+import base.SpecBase
+import models.UserAnswers
+import models.backend.PropertyDetails
+import org.mockito.ArgumentMatchers.{any, anyString}
+import org.mockito.Mockito.doNothing
+import org.mockito.MockitoSugar.when
+import org.scalatestplus.mockito.MockitoSugar.mock
+import play.api.inject.bind
+import play.api.mvc.Call
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import service.{BusinessService, PropertySubmissionService}
+import viewmodels.govuk.SummaryListFluency
+import views.html.foreign.allowances.ForeignAllowancesCheckYourAnswersView
+
+import java.time.LocalDate
+import scala.concurrent.Future
+
+class ForeignAllowancesCheckYourAnswersControllerSpec extends SpecBase with SummaryListFluency {
+
+  val taxYear = LocalDate.now.getYear
+  val countryCode = "aus"
+
+  val onwardRoute: Call = Call(
+    "GET",
+    s"/update-and-submit-income-tax-return/property/$taxYear/foreign-property/allowances/$countryCode/complete-yes-no"
+  )
+
+  val zeroEmissionCarAllowanceValue = 300
+  val zeroEmissionCarAllowance = BigDecimal.valueOf(zeroEmissionCarAllowanceValue)
+
+  val zeroEmissionGoodsVehicleAllowanceValue = 400
+  val zeroEmissionGoodsVehicleAllowance = BigDecimal.valueOf(zeroEmissionGoodsVehicleAllowanceValue)
+
+  val replacementOfDomesticGoodsValue = 600
+  val replacementOfDomesticGoods = BigDecimal.valueOf(replacementOfDomesticGoodsValue)
+
+  val annualInvestmentAllowanceSummaryValue = 100
+  val annualInvestmentAllowanceSummary = BigDecimal.valueOf(annualInvestmentAllowanceSummaryValue)
+
+  val otherCapitalAllowanceValue = 700
+  val otherCapitalAllowance = BigDecimal.valueOf(otherCapitalAllowanceValue)
+
+
+  "Check Your Answers Controller" - {
+
+    "must return OK and the correct view for a GET" in {
+
+      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers), true).build()
+
+      running(application) {
+        val request = FakeRequest(GET, routes.ForeignAllowancesCheckYourAnswersController.onPageLoad(taxYear, countryCode).url)
+
+        val result = route(application, request).value
+
+        val view = application.injector.instanceOf[ForeignAllowancesCheckYourAnswersView]
+        val list = SummaryListViewModel(Seq.empty)
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(list, taxYear, countryCode)(request, messages(application)).toString
+      }
+    }
+
+    "must redirect to Journey Recovery for a GET if no existing data is found" in {
+
+      val application = applicationBuilder(userAnswers = None, true).build()
+
+      running(application) {
+        val request = FakeRequest(GET, routes.ForeignAllowancesCheckYourAnswersController.onPageLoad(taxYear, countryCode).url)
+
+        val result = route(application, request).value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual controllers.routes.JourneyRecoveryController.onPageLoad().url
+      }
+    }
+//
```

Comment:
Why are these tests commented ?

Reply:
they are from ONSUBMIT method, As I have not implemented it, I commented it!